plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -91,7 +92,12 @@ public HudiSplitSource(
                 queue,
                 new BoundedExecutor(executor, getSplitGeneratorParallelism(session)),
                 createSplitWeightProvider(session),
-                partitions);
+                partitions,
+                throwable -> {
+                    trinoException.compareAndSet(null, new TrinoException(GENERIC_INTERNAL_ERROR,
+                            "Failed to generate splits for " + table.getTableName(), throwable));
+                    queue.finish();
```

Comment:
maybe call close()

and make close() call `splitLoaderFuture.cancel`

Reply:
call close on queue ? I thought finish will handle it @findepi  . could you please help me to understand this more

Reply:
> call close on queue ?

not on `queue`, on `this`

Reply:
@findepi  done .

Reply:
splitLoaderFuture.cancel on close is still missing @ryadav-uptycs